libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -179,6 +181,65 @@ bool IRGeneratorForStatements::visit(FunctionCall const& _functionCall)
 	return false;
 }
 
+bool IRGeneratorForStatements::visit(InlineAssembly const& _inlineAsm)
+{
+	auto const& loc = _inlineAsm.location();
+
+	solAssert(loc.start != 1 && loc.end != -1, "Expected source location");
+
+	vector<string> params, retParams, args;
+	// Prepare parameters and arguments
+	for (auto const& pair: _inlineAsm.annotation().externalReferences)
+	{
+		auto varDecl = dynamic_cast<VariableDeclaration const*>(pair.second.declaration);
+		solUnimplementedAssert(!!varDecl, "");
+
+		args.emplace_back(m_context.variableName(*varDecl));
+		params.emplace_back(varDecl->name());
+		retParams.emplace_back("ret_" + varDecl->name());
+	}
+
+	// Skip the "assembly" literal
+	int blockStart = loc.start + string("assembly").size();
+
+	stringstream funcName, assignVars;
+
+	funcName <<	"inline_assembly_" << loc.start << "_" << loc.end;
+
+	for (size_t i = 0; i < retParams.size(); i++)
+		assignVars << retParams[i] << " := " << params[i] << "\n";
```

Comment:
I think I can get rid of this part by just doing `$retParams := $params` in the `inlineAsm` string

Reply:
hmm or not.